spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java

```diff
@@ -781,17 +622,895 @@ private ProductDTO createProductWithProvidedAndDerivedProduct(String ownerKey) {
+                .containsAll(expectedPids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedPids));
+        }
+
+        @Test
+        public void shouldAllowFilteringWithActiveIncluded() {
+            // This is effectively the same as no filter -- we expect everything within the org back
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedPids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .map(ProductInfo::id)
+                .toList();
+
+            List<ProductDTO> output = this.adminClient.ownerProducts()
+                .getProductsByOwner(owner.getKey(), null, null, INCLUSION_INCLUDE, INCLUSION_INCLUDE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ProductDTO::getId)
+                .containsAll(expectedPids);
+        }
+
+        @Test
+        public void shouldAllowFilteringWithActiveExcluded() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedPids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(pinfo -> !pinfo.active())
+                .map(ProductInfo::id)
+                .toList();
+
+            List<ProductDTO> output = this.adminClient.ownerProducts()
+                .getProductsByOwner(owner.getKey(), null, null, INCLUSION_EXCLUDE, INCLUSION_INCLUDE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ProductDTO::getId)
+                .containsAll(expectedPids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedPids));
+        }
+
+        @Test
+        public void shouldAllowFilteringWithActiveExclusive() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedPids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(ProductInfo::active)
+                .map(ProductInfo::id)
+                .toList();
+
+            List<ProductDTO> output = this.adminClient.ownerProducts()
+                .getProductsByOwner(owner.getKey(), null, null, INCLUSION_EXCLUSIVE, INCLUSION_INCLUDE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ProductDTO::getId)
+                .containsAll(expectedPids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedPids));
+        }
+
+        @Test
+        public void shouldErrorWithInvalidActiveInclusion() {
+            OwnerDTO owner = this.owners.get(1);
+
+            assertBadRequest(() -> this.adminClient.ownerProducts()
+                .getProductsByOwner(owner.getKey(), null, null, "invalid_type", INCLUSION_INCLUDE));
+        }
+
+        @Test
+        public void shouldAllowFilteringWithCustomIncluded() {
```

Comment:
This one

Reply:
These being effectively duplicates is fine. When it comes to writing tests and verifying behavior, you don't want to combine separate validations as you end up losing that in-code contract should test implementations change in the future.

To some extent, this is along the lines of self-documenting code; but we're leaning more on proper test isolation and verifying a specification.